lib/abstract-generator.js

```diff
@@ -673,6 +679,29 @@ class AbstractGenerator {
     return a[0].substr(1, a[0].length - 2);
   }
 
+  /**
+   * This function replaces all external references in a given string.
+   * @param {string} str - The string in which the external references have to be replaced with their values.
+   */
+  _replaceExternalReferences(str) {
+    str =  '' + str;
+    const refs = str.match(/\{([^\}]*)\}/g);
+
+    if (refs) {
+      for (const ref of refs) {
+        if (ref.startsWith('{_')) {
+          const refWithoutBrackets = ref.substr(2, ref.length - 3);
+          const externalRefValue = this.externalReferences[refWithoutBrackets];
```

Comment:
What if the external reference isn't defined? if not, I'd fall back to keeping it a reference (so basically prefix with `\` here) and log it (and as such, we're not creating a breaking change, which we could deprecate in the future)

Reply:
Good point!